src/menu.c

```diff
@@ -0,0 +1,71 @@
+
+
+#include "../include/menu.h"
+
+menu_t mmenu[] = {'1', "Mergesort", '2', "Quicksort", '3', "Bubblesort", '4', "Insertionsort", 'M', "Show Menu", 'X', "Exit Program"};
+
+/** Function delivers list of menu points */
+void show_menu() {
+    int nr_menu_points = sizeof(mmenu)/sizeof(menu_t);
+    printf("Please choose from the following options\n");
+    //printf("Please choose the sorting algorithm you want to use:\n");
+    printf("----------------------------------------\n");
+    for (int i = 0; i<nr_menu_points; i++) {
+        printf("\t(%c) %s\n", mmenu[i].symbol, mmenu[i].text);
+    }
+    printf("----------------------------------------\n");
+}
+
+/** Function parses user input on stdin */
+void parse_input() {
+	char input = 0;
+	while (1) {
+		input = getchar();
+		while (getchar() != '\n');  //delete input buffer
+		switch (tolower(input)) {
+		case '1':
+			printf("Your choice: #%c\n", input);
+            //result = msort(number_a, number_b);  //include functions here for sorting algorithms and outputs
+            //printf("bla : %f\n", result);
+			break;
+		case '2': 
+			printf("Your choice: #%c\n", input);
+			break;
+		case '3': 
+			printf("Your choice: #%c\n", input);
+			break;
+		case '4': 
+			printf("Your choice: #%c\n", input);
+			break;
+            /* // check return value for errors
+			if (input == -1) {
+				break;
+			}
+			break; */
+		case 'm':
+            show_menu();
+            continue;
+		case 'x':
+			printf("Closing program...\n\n");
+            exit(EXIT_SUCCESS);
+		default:
+			printf("Please use ""'M'"" (for menu) to show viable options.\n");
+			break;
+		}
+	} //end while
+}
+
+/** MAIN ROUTINE*************************************************************/
+int main(void) {
+    printf("\nProgram starting...\n");
+    show_menu();
+
+/* Example for debugging */
+if (D) {
```

Comment:
hier die testweise DEBUG IFs

Reply:
Hab grade herum recherchiert und gefunden dass viele standardmäßig auch diese Ausführung verwenden:
```
#define DEBUG 1
....
#if DEBUG

#endif
```
Der Unterschied ist dass mit #if DEBUG schon während dem compiler Prozess das #if evaluiert wird und bei if() bei jedem neuen Zyklus verarbeitet wird -> das heißt es ist ein Tick schneller. Für unsere Anwendungen passt dass if() aber komplett. Aber nur good to know :)

Reply:
Hi, danke fürs Checken. Dann machen wir es gleich richtig, werde ich ändern.

Reply:
noch ne Frage: schieben wir dieses Define dann am Besten auch in eine main.h (Header für main)? Wäre jetzt meine bevorzugte Variante und baue ich dann gleich auch ein...

Reply:
Ja Passt :)